### PR TITLE
Merge from AK -> CCS

### PR DESCRIPTION
**Pull request by lbradstreet**
Merge branch 'trunk' into sync-13-jan-2020
    
    Conflicts:
            build.gradle
            core/src/test/scala/unit/kafka/cluster/PartitionTest.scala
            docs/upgrade.html
            gradle/dependencies.gradle
            streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala

Conflicts are primarily due to revert the commit that dropped scala 2.11 https://github.com/confluentinc/kafka/commit/98d606115af89a7e258054126fcbf2b2b479991b.

I also had to fix some new tests for compiling under scala 2.11 https://github.com/confluentinc/kafka/pull/246/commits/b83a69596617fa34e1a6bfa8c1674dff4ad60c38.

### CODE REVIEW

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala [hunk 1]
```diff
@@ -31,17 +31,15 @@ import org.apache.kafka.streams.kstream.{
...
-  Transformer,
-  TransformerSupplier,
-  ValueJoiner,
-  ValueMapper,
-  StreamJoined => StreamJoinedJ,
   KGroupedStream => KGroupedStreamJ,
   KStream => KStreamJ,
   KTable => KTableJ,
-  Materialized => MaterializedJ
+  Materialized => MaterializedJ,
```
Comment (lbradstreet) [inline streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: Merge conflict here

File: streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala [hunk 2]
```diff
@@ -144,6 +141,93 @@ class TopologyTest {
     assertEquals(getTopologyScala, getTopologyJava)
   }
 
+  @Test def shouldBuildIdenticalTopologyInJavaNScalaCogroupSimple(): Unit = {
```
Comment (lbradstreet) [inline streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala]: Merge conflict here

File: core/src/test/scala/unit/kafka/cluster/PartitionTest.scala [hunk 3]
```diff
@@ -954,7 +955,7 @@ class PartitionTest extends AbstractPartitionTest {
       // Append records to partitions, one partition-per-thread
       val futures = partitions.map { partition =>
         executor.submit(CoreUtils.runnable {
-          (1 to 10000).foreach { _ => partition.appendRecordsToLeader(createRecords(baseOffset = 0), isFromClient = true) }
+          (1 to 10000).foreach { _ => partition.appendRecordsToLeader(createRecords(baseOffset = 0), origin = AppendOrigin.Client, requiredAcks = 0) }
```
Comment (lbradstreet) [inline core/src/test/scala/unit/kafka/cluster/PartitionTest.scala]: Merge conflict here due to upstream change that switches CoreUtils.runnable for a scala lambda used as a runnable.

File: docs/upgrade.html [hunk 4]
```diff
@@ -21,10 +21,18 @@
...
+        while it is in the middle of a rebalance for those partitions still owned by the consumer; in addition
+        <code>Consumer#commitSync</code> now may throw a non-fatal <code>RebalanceInProgressException</code> to notify
+        users of such an event, in order to distinguish from the fatal <code>CommitFailedException</code> and allow
+        users to complete the ongoing rebalance and then reattempt committing offsets for those still-owned partitions.</li>
     <li>For improved resiliency in typical network environments, the default value of
         <code>zookeeper.session.timeout.ms</code> has been increased from 6s to 18s and
         <code>replica.lag.time.max.ms</code> from 10s to 30s.</li>
-    <li>New DSL operator <code>cogroup()</code> has been added for aggregating multiple streams together at once</li>
+    <li>New DSL operator <code>cogroup()</code> has been added for aggregating multiple streams together at once.</li>
+    <li>Scala 2.11 is no longer supported. See
```
Comment (lbradstreet) [inline docs/upgrade.html]: FIXME. This line should have been dropped.

File: build.gradle [hunk 5]
```diff
@@ -789,7 +790,14 @@ project(':core') {
     testCompile libs.slf4jlog4j
     testCompile libs.jfreechart
   }
-  
+
+  scoverage {
```
Comment (lbradstreet) [inline build.gradle]: Conflict here

File: build.gradle [hunk 6]
```diff
@@ -36,7 +36,7 @@ buildscript {
     classpath "com.github.jengelman.gradle.plugins:shadow:$versions.shadowPlugin"
     classpath "org.owasp:dependency-check-gradle:$versions.owaspDepCheckPlugin"
     classpath "com.diffplug.spotless:spotless-plugin-gradle:$versions.spotlessPlugin"
-    classpath "gradle.plugin.com.github.spotbugs:spotbugs-gradle-plugin:$versions.spotbugsPlugin"
+    classpath "com.github.spotbugs:spotbugs-gradle-plugin:$versions.spotbugsPlugin"
```
Comment (lbradstreet) [inline build.gradle]: Merge conflict here

File: gradle/dependencies.gradle [hunk 7]
```diff
@@ -65,28 +65,28 @@ versions += [
   argparse4j: "0.7.0",
   avro: "1.9.1",
```
Comment (lbradstreet) [inline gradle/dependencies.gradle]: Merge conflict from here to jacksonDatabind.

**Comment by lbradstreet** [comment]
Failure in kafka.admin.TopicCommandWithAdminClientTest.testDescribeUnderReplicatedPartitionsWhenReassignmentIsInProgress. Rerunning.

**Comment by cmccabe** [comment]
The changes look pretty small.  LGTM

**Comment by cmccabe** [comment]
Do we know what has to be done to fully drop old scala support in ccloud?

I think we should prioritize fixing this.  It's going to get increasingly difficult to support scala 2.11 in ce-kafka if upstream doesn't support it.

**Comment by lbradstreet** [comment]
Failure in `ERROR: test-cp-downstream-builds #10145 completed with status FAILURE (propagate: false to ignore)`. Merging.

**Pull request merged**